### ISSUE TITLE
hipchat

### ISSUE THREAD

**Issue by leram84** [causefx/iDashboard-PHP#19]
I think i might have found a really good work around to not being able to iframe slack. Im setting up hipchat instead, and then using sameroom to sync with slack on my end. this way i keep using slack and only slack for all my stuff, and users can use hipchat in the guest idash. Only problem is, the log in. I've tried every combination of url i can, but the login page is a redirect from the chat homepage. https://mygroup.hipchat.com/chat will redirect to https://mygroup.hipchat.com/sign_in?d=%2Fchat in a normal web page, and then after you sign in it takes you back to https://mygroup.hipchat.com/chat. So if i set the tab to https://mygroup.hipchat.com/chat AND i'm already logged into hipchat in another tab in chrome, the tab loads up just like it should, right to the hipchat homepage. BUT if im not logged in, the page loads blank. I tried even setting the tab url to https://mygroup.hipchat.com/sign_in?d=%2Fchat but that just loads a blank page by default. I'm so close to making this work, i just need the tab to redirect to the login page INSIDE idashboard IF im not already logged in. Do you know if this can be done?

**Comment by causefx** [comment]
let me take a stab at it :)

**Comment by causefx** [comment]
:(
![image](https://cloud.githubusercontent.com/assets/16184466/20127231/e28ff5b0-a5f9-11e6-8e84-a5c8df35672b.png)

**Comment by causefx** [comment]
The only current workaround is for Chrome.  

Install this plugin:
https://chrome.google.com/webstore/detail/ignore-x-frame-headers/gleekbfjekiniecknbkamfmkohkpodhe

reload and done

**Comment by leram84** [comment]
wait, so JUST the sign-in page is blocked, but the rest of the site is fine.... wha??

**Comment by causefx** [comment]
yea could be.  Did you install the plug in?  or do you use firefox? 

Firefox: https://addons.mozilla.org/en-US/firefox/addon/ignore-x-frame-options/

**Comment by leram84** [comment]
no, i don't want to do anything that won't work on all platforms by default.... guess im back to the drawing board. good news is there's currently just over 700k messaging apps out there for me to try :/

**Comment by causefx** [comment]
I'll keep looking too.

**Comment by leram84** [comment]
after ruling out maybe 20-30 different chat apps, im inclined to roll my own. I know its overkill for something like this but i feel like it will be a fun learning experience so im diving in.... (not to mention the fact that if i can learn enough to get plexauth working in nginx, it should (in theory) be compatible). So far i think Mattermost seems to have the most potential, but im running into issues figuring out how to deploy it, mainly since my server is win10, and i don't want to add any crazy overhead to it just for something like this (not gonna install docker or any kind of linux vm on the pms). I was thinking about using one of my rpi3's i have lying around, and that seems like a pretty good idea, EXCEPT i have no clue how to integrate it into idash if i do go that route.... I'm hoping its as easy as reserving an ip for the pi and then creating a new location block in my nginx server with that ip, but i have no idea if that's how it works. Any thoughts on this?

And also, I think i may have asked you this before (or you might be the one that actually pointed me to it in the first place) but do you use plexauth with your setup? I'm curious if there would be a way to integrate some of the more complicated parts of the configuration directly into idashboard, since i can't think of a single use case for idash or muximux that wouldn't be improved with sso. I mean having a single sign in for admin would be awesome, having it for guests though.... THAT is cool as shit!

**Comment by causefx** [comment]
I haven't used Plex auth so I can comment on that. I will look into a chat type thing that we can incorporate. U want it for just chatting?

**Comment by causefx** [comment]
Did you ever find a good chat alternative?  I'm currently testing out Riot, https://goo.gl/NvrguK for troubleshooting iDash.  Not sure what to go with...

**Comment by leram84** [comment]
That's interesting, but I REALLY prefer the slack type apps. I have had some really good progress with mattermost, and I'm still pretty convinced its the best solution, but i've hit a bit of a wall. They do have a windows option now, so i was able to deploy it on my server with out issues (well some issues lol, but all worked out) My problem now is with serving it up via nginx. For some reason they don't support running via subgroups, so https://www.mydomain.com/mattermost is out. They only want it running from https://www.mydomain.com but im keeping that for idashboard, so im working on figuring out how to setup a virtual host in nginx. It's probably something really easy that most people won't have any trouble with, but every line of code in my nginx config was taken from guides, so I just don't know how yet. BUT i'm almost positive it will work, and when it does, i feel like that will be the most elegant solution. Using mattermost will let me create a username for each person i share plex with. Then i can let them use it to communicate with me/each other. I can also setup notifications for anything i want. I can even  (and this is probably the coolest thing) setup ifttt forwards for all the plexrequests emails to go to each users mattermost account. Basically this means I won't need to send emails to anyone, but everyone can instead just log into my domain, and then watch, request, and get notifications for when requests are filled, or reported content is fixed, on top of any of my own notifications... a little bit of overkill but a whole lot of awesome :))

**Comment by causefx** [comment]
I'll have to look into that.  I will spin up a VM tonight and get it going.  I will let you know how it goes :)

**Comment by causefx** [comment]
have u tried this in a tab lately? https://mygroup.hipchat.com/chat

**Comment by leram84** [comment]
doesn't work... set to sameorigin. Are you saying its working for you?

**Comment by causefx** [comment]
i dont have a login let me sign up

**Comment by causefx** [comment]
![image](https://cloud.githubusercontent.com/assets/16184466/21337258/12f01a10-c629-11e6-9e81-674eafc5a25e.png)

**Comment by causefx** [comment]
i'm an idiot, i forgot i have that plug installed

**Comment by leram84** [comment]
hahaha... oh yeah. 

You know what though... this might be yet another issues that would be solved with plexauth. I have no idea if this is true or not, but if it can work without actually needing the page to load in the iframe (as in, if the oauth bypasses the login page altogether) Then, it wouldn't matter that the login page doesn't work, since the rest of the site seems to run smoothly in a tab once the machine is logged in.

**Comment by causefx** [comment]
PlexAuth doesn't work like that.  It runs along with Nginx.  So from what I understand, I just allows whoever is on your plex shares to have access to whatever pages on your server that are protected by nginx.

It authentication for your domain only.

**Comment by leram84** [comment]
That's what i thought initially [but asked the dev and he said it does actually log you into apps (like plexpy) that support multiple users.](https://github.com/hjone72/PlexAuth/issues/4)

Unless i totally misunderstood him...

**Comment by hjone72** [comment]
@leram84, you have misunderstood slightly. I have modified my PlexPy to work with PlexAuth. Most apps require some sort of modification for PlexAuth to work with them.
I don't believe there is a solution to `X-Frame-Options` as it is specifically designed to prevent click jacking.

**Comment by leram84** [comment]
@hjone72 huh... ok, well if that's the case i don't think ill be able to use plexauth at all. As it stands my main use case for it was to allow my users to get guest (single sign on) access to Plex, ombi, tumblr, and mattermost (and trakt, when they get around to enabling iframes... something they said they are looking into), through my guest organizr page, and im nowhere near competent enough to fork all those.

But thanks for responding here! It helps a lot knowing what is and isn't possible as i work my way through all this!

**Comment by hjone72** [comment]
Yeah, unfortunately it definitely isn't capable of doing all that. As @causefx quite rightly said, it is auth for your domain and the apps that you host. Good luck with your mission though! 👍

**Comment by causefx** [comment]
@hjone72 thanks for clearing that up brother.  I did just take another peak at your project.  That PlexUser class is impressive.  I think that is something I may want to look into later to create the plex login into these type of dashboards.  thanks for doing what you do.
